src/bridges/convex/ConvexStakingBridge.sol

```diff
@@ -0,0 +1,265 @@
+    */
+    function _claimSubsidy(
+        ISubsidy _subsidy,
+        AztecTypes.AztecAsset calldata _inputAssetA,
+        AztecTypes.AztecAsset calldata _inputAssetB,
+        AztecTypes.AztecAsset calldata _outputAssetA,
+        AztecTypes.AztecAsset calldata _outputAssetB,
+        address _rollupBeneficiary
+    ) internal {
+        _subsidy.claimSubsidy(
+            computeCriteria(_inputAssetA, _inputAssetB, _outputAssetA, _outputAssetB, 0),
+            _rollupBeneficiary
+        );
+    }
+
+    /** 
+    @notice Internal function to deposit Curve LP tokens.
+    @notice RCT is minted for the bridge. Mirrors balance of minted Convex token for the bridge.
+    */
+    function _deposit(
+        AztecTypes.AztecAsset memory _outputAssetA,
+        uint256 _totalInputValue,
+        PoolInfo memory _selectedPool
+    ) internal returns (uint256 outputValueA) {
+        uint256 startCurveRewards = ICurveRewards(_selectedPool.curveRewards).balanceOf(address(this));
+
+        BOOSTER.deposit(_selectedPool.poolId, _totalInputValue, true);
+
+        uint256 endCurveRewards = ICurveRewards(_selectedPool.curveRewards).balanceOf(address(this));
+
+        outputValueA = (endCurveRewards - startCurveRewards);
+
+        IRepConvexToken(_outputAssetA.erc20Address).mint(_totalInputValue);
+    }
+
+    /** 
+    @notice Internal function to withdraw Curve LP tokens.
+    @notice RCT is burned for the bridge. Mirrors balance of minted Convex token for the bridge.
+    */
+    function _withdraw(
+        AztecTypes.AztecAsset memory _inputAssetA,
+        AztecTypes.AztecAsset memory _outputAssetA,
+        uint256 _totalInputValue,
+        PoolInfo memory _selectedPool
+    ) internal returns (uint256 outputValueA) {
+        uint256 startCurveLpTokens = ICurveLpToken(_outputAssetA.erc20Address).balanceOf(address(this));
+
+        // transfer ownership of Convex tokens from CrvRewards back to the bridge
+        ICurveRewards(_selectedPool.curveRewards).withdraw(_totalInputValue, true); // always claim rewards
+
+        BOOSTER.withdraw(_selectedPool.poolId, _totalInputValue);
+
+        uint256 endCurveLpTokens = ICurveLpToken(_outputAssetA.erc20Address).balanceOf(address(this));
+
+        outputValueA = (endCurveLpTokens - startCurveLpTokens);
+
+        IRepConvexToken(_inputAssetA.erc20Address).burn(_totalInputValue);
+    }
+
+    /**
+     * @notice Computes the criteria that is passed when claiming subsidy.
+     * @param _inputAssetA The input asset
+     * @param _outputAssetA The output asset
+     * @return The criteria
+     */
+    function computeCriteria(
+        AztecTypes.AztecAsset calldata _inputAssetA,
+        AztecTypes.AztecAsset calldata,
+        AztecTypes.AztecAsset calldata _outputAssetA,
+        AztecTypes.AztecAsset calldata,
+        uint64
+    ) public view override(BridgeBase) returns (uint256) {
+        if (deployedClones[_inputAssetA.erc20Address] == _outputAssetA.erc20Address) {
```

Comment:
Yes

Reply:
Okok, well, we've usually assigned it a different criteria since it might be desirable to subsidize for example only deposit. For this reason please simplify it and make it return different ones.